RFCs/0014-ToJsonString.md

```diff
@@ -0,0 +1,124 @@
+# New Intrinsic Function to Convert Template Block to JSON String
+
+* **Original Author(s):**: @mluk-aws
+* **Tracking Issue**: [Tracking Issue](https://github.com/aws-cloudformation/cfn-language-discussion/issues/14)
+
+# Summary
+
+We will support an intrinsic function called `Fn::ToJsonString` that enables developers to convert a template block into an escaped JSON string, which can be used as input values to string-type properties of CloudFormation resources.
+
+# Examples
+
+Here is an example of a resource containing a string-type property using a JSON formatted string.
+```
+"MySecret": {
+    "Type": "AWS::SecretsManager::Secret",
+    "Properties": {
+        "Name": "MySecret",
+        "SecretString": "{\"password\":\"Pa$$word\",\"secretToken\":\"123\"}"
+    }
+}
+```
+
+With the new `Fn::ToJsonString` intrinsic function we can simplify to the following:
+
+## JSON
+```
+"MySecret": {
+    "Type": "AWS::SecretsManager::Secret",
+    "Properties": {
+        "Name": "MySecret",
+        "SecretString": {
+            "Fn::ToJsonString": {
+                "password":"Pa$$word",
+                "secretToken": "123"
+            }
+        }
+    }
+}
+```
+
+## YAML
+```
+MySecret:
+    Type: AWS::SecretsManager::Secret
+    Properties:
+        Name: MySecret
+        SecretString:
+            Fn::ToJsonString:
+                password: "Pa$$word"
+                secretToken: "123"
+```
+
+### Short Form Syntax
+```
+MySecret:
+    Type: AWS::SecretsManager::Secret
+    Properties:
+        Name: MySecret
+        SecretString:
+            !ToJsonString:
+                password: "Pa$$word"
+                secretToken: "123"
+```
+
+# Motivation
+
+A CloudFormation user may want to use JSON strings as input to a resource property. For example, `AWS::CloudWatch::Dashboard` requires a JSON string for the `DashboardBody` attribute. A user may even want to use a JSON string as input to attributes where only a general string is required, such as in `SecretString` of `AWS::SecretsManager::Secret`.
```

Comment:
Like the dashboard example. Don't really like the SecretsManager example since, again, hardcoding secrets in your template is bad practice.